### PR TITLE
Microdata test

### PR DESCRIPTION
**Pull request by RomanFedytskyi**

### CODE REVIEW

File: tests/testmicrodata.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * @fileoverview Defines tests for microdata widget.
+ * success criterias:
+   - 'microdata-container' shoud exist and contains ul>li;
+   - 'widget-subheader' class shuold have 'Found Number errors and
+     Number warnings out of Number items.' expression;
+   - if widget is empty 'no-data' class should exist;
+ */
+
+
+/** @const {!RegExp} */ var PATTERN = new RegExp('Found\\s[0-9]*\\serrors' +
```
Comment (vpodk) [inline tests/testmicrodata.js]: Please do not hardcode localized text into regular expression.

File: tests/testmicrodata.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  var subheader = widget.querySelector('.widget-subheader').textContent;
+  /** @type {Node} */
+  var content = widget.querySelector('.widget-content > ul');
+  /** @type {Node} */
+  var list = content.getElementsByTagName('ul');
+  /** @type {Node} */
+  var listItems = content.querySelector('ul > li');
+  /** @type {Node} */
+  var noData = widget.querySelector('.no-data');
+  subheader = subheader.replace(/\s\s+/g, ' ');
```
Comment (vpodk) [inline tests/testmicrodata.js]:
``` javascript
subheader = subheader.replace(/\s+/g, ' ');
```

File: tests/testmicrodata.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  if (!content) {
+    console.log('1');
```
Comment (vpodk) [inline tests/testmicrodata.js]: Please remove debugging code.

File: tests/testmicrodata.js (hunk 2 above)
Comment (RomanFedytskyi) [inline tests/testmicrodata.js]: fixed

File: tests/testmicrodata.js (hunk 3 above)
Comment (RomanFedytskyi) [inline tests/testmicrodata.js]: fixed

File: tests/testmediaqueries.js [hunk 4]
```diff
@@ -1,9 +1,13 @@
 /**
  * @fileoverview Defines tests for testMediaQueries.
+ * success criterias:
```
Comment (vpodk) [inline tests/testmediaqueries.js]: https://en.wikipedia.org/wiki/Sentence_(linguistics)

File: tests/testmicrodata.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+  if (content) {
+    if (listItems) {
+      return false;
+    } else if (!noData) {
+      return true;
+    }
+  }
+
+  return false;
```
Comment (vpodk) [inline tests/testmicrodata.js]:
``` javascript
/** @type {boolean} */ var result = false;
...
return result; // Only one return statement.
```

File: tests/testmediaqueries.js (hunk 4 above)
Comment (RomanFedytskyi) [inline tests/testmediaqueries.js]: fixed

File: tests/testmicrodata.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+/** @const {!RegExp} */ var PATTERN = new RegExp('\\w*\\s[0-9]*\\s\\w*\\s\\w*' +
+    '\\s[0-9]*\\s(\\w*\\s){3}[0-9]*\\s\\w*\\.', 'gmi');
+
+
+/**
+ * @return {boolean} Returns "true" if test failed.
+ */
+function testMicrodata() {
+  /** @type {boolean} */ var result =false;
```
Comment (mshchygol) [inline tests/testmicrodata.js]: Please use space after '='

**Comment by mshchygol** [comment]
LGTM 👍

**Pull request closed without merging**